packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/dropNode/useDropNode.ts

```diff
@@ -190,23 +176,30 @@ export const useDropNode = (): UseDropNodeValue => {
       compatibleNodeIds: compatibleNodes,
       droppableOnDiagram: dropDataEntry?.droppableOnDiagram || false,
     });
-  };
-
-  const onNodeDrag: NodeDragHandler = (_event, node) => {
-    if (draggedNode && !draggedNode.data.isBorderNode) {
-      const intersections = getIntersectingNodes(node).filter((intersectingNode) => !intersectingNode.hidden);
-      const newParentId =
-        [...intersections]
-          .filter((intersectingNode) => !isDescendantOf(draggedNode, intersectingNode, getNodeById))
-          .sort((n1, n2) => getNodeDepth(n2, intersections) - getNodeDepth(n1, intersections))[0]?.id || null;
-      setTargetNodeId(newParentId);
-    }
-  };
+  }, []);
+
+  const onNodeDrag: NodeDragHandler = useCallback(
+    (_event, node) => {
+      if (draggedNode && !draggedNode.data.isBorderNode) {
+        const intersections = getIntersectingNodes(node).filter((intersectingNode) => !intersectingNode.hidden);
+        const newParentId =
+          [...intersections]
+            .filter((intersectingNode) => !isDescendantOf(draggedNode, intersectingNode, getNodeById))
+            .sort((n1, n2) => getNodeDepth(n2, intersections) - getNodeDepth(n1, intersections))[0]?.id || null;
+        setTargetNodeId(newParentId);
+      }
+    },
+    [draggedNode?.id]
+  );
 
+  const reactflow = useReactFlow<NodeData, EdgeData>();
   const onNodeDragStop: (onDragCancelled: (node: Node) => void) => NodeDragHandler = useCallback(
     (onDragCancelled) => {
       return (event, _node) => {
-        const dropPosition = computeDropPosition(event, viewport, element);
+        const dropPosition = reactflow.screenToFlowPosition({
```

Comment:
I have not tested that yet but since the other code had a bug, I have the feeling that it does not what you think it does.